### ISSUE TITLE
[Nakamoto] Support `TenureChange` Stacks transaction

### ISSUE THREAD

**Issue by zone117x** [hirosystems/stacks-blockchain-api#1733]
This is likely the first new tx type that will be implemented for Nakamoto.

See stacks-blockchain issue: https://github.com/stacks-network/stacks-blockchain/issues/3944

Corresponding WIP stacks-blockchain PR: https://github.com/stacks-network/stacks-blockchain/pull/3974

**Comment by zone117x** [comment]
Potentially unblocked by https://github.com/stacks-network/stacks-blockchain/pull/4001

**Comment by zone117x** [comment]
Deserialization for `TenureChange` tx types ready over at https://github.com/hirosystems/stacks-encoding-native-js/pull/14

**Comment by zone117x** [comment]
Implemented in https://github.com/hirosystems/stacks-blockchain-api/pull/1753

There's likely going to be another change to the wire format for this tx. Will re-open or open a new issue if/when that happens.
